Does not work with 5.4

Thanks for your project. I tried to compile it with the latest version of lua and it doesn't work due to this [changes](https://www.lua.org/manual/5.4/manual.html#8.3)

```
/mnt/vault/tmp/github/jordanvrtanoski/luacpp/Source/Registry/LuaCompiler.cpp:39:9: error: use of undeclared identifier 'LUA_ERRGCMM'
                        case LUA_ERRGCMM:
                             ^
1 error generated.
```

Since the project was created recently I assumed it was using the latest (or recent) version of lua. I think the main README should at the very minimum states which version of lua you are targeting.

--- a/Source/Registry/LuaCompiler.cpp
+++ b/Source/Registry/LuaCompiler.cpp
@@ -36,9 +36,11 @@ void inline _checkErrorAndThrow(LuaState &L, int error) {
 			case LUA_ERRMEM:
 				throw std::runtime_error("Out of memory");
 				break;
+#if LUA_VERSION_NUM < 504
 			case LUA_ERRGCMM:
 				throw std::out_of_range("GC Error while loading");
 				break;
+#endif
 			case LUA_ERRSYNTAX:
 				throw std::logic_error(lua_tostring(L,1));
 				break;

--- a/Source/UnitTest/TestLuaCompiler.cpp
+++ b/Source/UnitTest/TestLuaCompiler.cpp
@@ -75,7 +75,9 @@ namespace LuaCpp {
 		EXPECT_NO_THROW(_checkErrorAndThrow(*L, LUA_OK));
 		EXPECT_THROW(_checkErrorAndThrow(*L, LUA_ERRMEM), std::runtime_error);
 		lua_pushstring(*L, "some error");
+#if LUA_VERSION_NUM < 504
 		EXPECT_THROW(_checkErrorAndThrow(*L, LUA_ERRGCMM), std::out_of_range);
+#endif
 		EXPECT_THROW(_checkErrorAndThrow(*L, LUA_ERRSYNTAX), std::logic_error);
 
 		EXPECT_THROW(_checkErrorAndThrow(*L, 9999), std::runtime_error);

